spec/lib/rbac/filterer_spec.rb

```diff
@@ -765,29 +765,73 @@ def combine_filtered_ids(user_filtered_ids, belongsto_filtered_ids, managed_filt
+                it "private cloud templates" do
+                  User.current_user = user_2
+                  results = described_class.filtered(TemplateCloud, :user => user_2)
+                  expect(results).to match_array([cloud_template_root])
+                end
+              end
+            end
           end
 
-          context "when user is restricted user" do
-            let(:tenant_2) { FactoryGirl.create(:tenant, :parent => default_tenant, :source_type => 'CloudTenant') } # T2
-            let(:group_2) { FactoryGirl.create(:miq_group, :tenant => tenant_2) } # T1
-            let(:user_2) { FactoryGirl.create(:user, :miq_groups => [group_2]) }
-            let(:tenant_3) { FactoryGirl.create(:tenant, :parent => tenant_2) } # T3
-            let!(:cloud_template) { FactoryGirl.create(:template_cloud, :tenant => tenant_3, :publicly_available => true) }
+          context "when tenant has source_id" do
+            let!(:cloud_template_2) { FactoryGirl.create(:template_cloud, :tenant => tenant_2, :publicly_available => false) }
 
-            it "returns all public cloud templates" do
+            it "finds tenant's private cloud templates" do
+              User.current_user = user_2
+              tenant_2.source_id = 1
               results = described_class.filtered(TemplateCloud, :user => user_2)
-              expect(results).to match_array([cloud_template, cloud_template_root])
+              expect(results).to match_array([cloud_template_2])
             end
 
-            context "should ignore" do
-              let!(:cloud_template) { FactoryGirl.create(:template_cloud, :tenant => tenant_3, :publicly_available => false) }
-              it "private cloud templates" do
-                results = described_class.filtered(TemplateCloud, :user => user_2)
-                expect(results).to match_array([cloud_template_root])
-              end
+            let(:cloud_template_3) { FactoryGirl.create(:template_cloud, :tenant => tenant_2, :publicly_available => true) }
+
+            it "finds tenant's private and public cloud templates" do
+              User.current_user = user_2
+              tenant_2.source_id = 1
+              results = described_class.filtered(TemplateCloud, :user => user_2)
+              expect(results).to match_array([cloud_template_2, cloud_template_3])
+            end
+
+            let(:cloud_template_4) { FactoryGirl.create(:template_cloud, :tenant => default_tenant, :publicly_available => false) }
+
+            it "ignores other tenant's private templates" do
+              User.current_user = user_2
+              tenant_2.source_id = 1
+              results = described_class.filtered(TemplateCloud, :user => user_2)
+              expect(results).to match_array([cloud_template_2, cloud_template_3])
+            end
+
+            let(:cloud_template_5) { FactoryGirl.create(:template_cloud, :tenant => default_tenant, :publicly_available => true) }
```

Comment:
it have to be evaluated before RBAC is called. And can you put it together after line context "when tenant has source_id "
```
            let!(:cloud_template_2) { FactoryGirl.create(:template_cloud, :tenant => tenant_2, :publicly_available => false) }
            let!(:cloud_template_3) { FactoryGirl.create(:template_cloud, :tenant => tenant_2, :publicly_available => true) }
            let!(:cloud_template_4) { FactoryGirl.create(:template_cloud, :tenant => default_tenant, :publicly_available => false) }
            let!(:cloud_template_5) { FactoryGirl.create(:template_cloud, :tenant => default_tenant, :publicly_available => true) }
```